debezium-core/src/main/java/io/debezium/connector/base/ChangeEventQueue.java

```diff
@@ -126,9 +139,13 @@ public void enqueue(T record) throws InterruptedException {
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("Enqueuing source record '{}'", record);
         }
-
+        // Waiting for queue to add more record.
+        while (currentQueueSizeInBytes.get() > maxQueueSizeInBytes) {
+            Thread.sleep(5000);
```

Comment:
Instead of hardoced value I recommend to reuse `pollInterval`

Reply:
Thanks! I will update it